runtime/jvmti/jvmtiStartup.c

```diff
@@ -185,25 +202,36 @@ createAgentLibraryWithOption(J9JavaVM *vm, J9VMInitArgs *argsList, IDATA agentIn
 
 /**
  * Create an agent library from an option index.
+ * Three agent options are expected: VMOPT_AGENTLIB_COLON, VMOPT_AGENTPATH_COLON or MAPOPT_XRUNJDWP.
  *
  * @param[in] vm Java VM
  * @param[in] argsList a J9VMInitArgs
- * @param[in] agentColon the agent option, VMOPT_AGENTLIB_COLON, VMOPT_AGENTPATH_COLON or MAPOPT_XRUNJDWP
  * @param[in] loadLibrary indicate if the agent library created to be loaded or not
  * @param[in] isXrunjdwp indicate if the agent option is MAPOPT_XRUNJDWP
+ * @param[in] isAgentPath indicate if the agent option is VMOPT_AGENTPATH_COLON
  *
  * @return TRUE if succeeded, otherwise FALSE
  */
 static BOOLEAN
-processAgentLibraryFromArgsList(J9JavaVM *vm, J9VMInitArgs *argsList, const char *agentColon, BOOLEAN loadLibrary, BOOLEAN isXrunjdwp)
+processAgentLibraryFromArgsList(J9JavaVM *vm, J9VMInitArgs *argsList, BOOLEAN loadLibrary, BOOLEAN isXrunjdwp, BOOLEAN isAgentPath)
 {
-	IDATA agentIndex = FIND_AND_CONSUME_ARG_FORWARD(argsList, STARTSWITH_MATCH, agentColon, NULL);
+	IDATA agentIndex = 0;
 	BOOLEAN result = TRUE;
+	const char *agentColon = NULL;
 
+	if (isXrunjdwp) {
+		agentColon = MAPOPT_XRUNJDWP;
+	} else if (isAgentPath) {
+		agentColon = VMOPT_AGENTPATH_COLON;
+	} else {
+		/* only three options are expected */
+		agentColon = VMOPT_AGENTLIB_COLON;
+	}
```

Comment:
A better representation of three possibilities would be a 3-value enum.

Reply:
Added an `enum` `OPTIONS_JDWP_AGENT` for these three JDWP options.